txlib/api/resources.py

```diff
@@ -24,19 +25,59 @@ class Resource(BaseModel):
     }
     url_fields = {'project_slug', 'slug'}
 
+    BINARY_TYPES = ['DOCX', 'XLSX']
+
+    def _is_binary_file(self):
+        if hasattr(self, 'i18n_type'):
+            return self.i18n_type in self.BINARY_TYPES
+        return False
+
     def retrieve_content(self):
         """Retrieve the content of a resource."""
         path = self._construct_path_to_source_content()
         res = self._http.get(path)
         self._populated_fields['content'] = res['content']
         return res['content']
 
+    def _act_on_temp_file(self, action, path, file_content, kwargs):
+        """Wrap a file action (callable) using a temporary file that is disposed of automatically on completion."""
+        with tempfile.NamedTemporaryFile(prefix=self.name) as temp_file:
+            temp_file.write(file_content)
+            # rewind file after writing so it can be read()
+            temp_file.seek(0)
+            response = action(path, kwargs, temp_file.name)
```

Comment:
This line is not so clear, I did not know why `temp_file.name` was a valid argument until I had a look at `http_requests.py`, seeing that the `put` function had a `filename=None` argument in it. The same applies for `create`, but not for `delete` or `get`, so not every action is valid in our case.

I believe it is worth spending some time explaining the reasoning behind this.